core/controllers/editor.py

```diff
@@ -701,9 +701,15 @@ def get(self, exploration_id, exploration_version):
         except:
             raise self.PageNotFoundException
 
-        self.render_json(stats_services_old.get_exploration_stats(
-            current_exploration.id, exploration_version,
-            current_exploration.states))
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
+            if exploration_version == 'all':
+                exploration_version = current_exploration.version
```

Comment:
Not sure I understand  this. My understanding is that "all" represents stats aggregated across all versions, and a particular version number represents only stats for a specific version. This seems to be doing something different though. Could you please clarify?

Reply:
@seanlip This was something we never took into account in this framework - we never did plan to have version specific aggregation of stats. Given a version number, we retrieve the ExplorationStatsModel which represents all the aggregated stats from exploration creation until the specified version number. So, the concept of version specific stats is sort of lost here.

If the version specific aggregation is really necessary, I think we can add on to this framework, a method that would retrieve the ExplorationStatsModel for the specified version and the immediate previous version, subtract their values and return it.

Reply:
I'm fine not having it, but in that case why pass exp_version? Shouldn't it be just get_exploration_stats_by_id(exp_id)?

Reply:
@seanlip I thought about that. But, we need the exp_version also to retrieve from data store since we are keeping around all the older version's stats models as well.
Inside stats_services, we can't import exp_services. So, we cannot do something like : exp_services.get_exploration_by_id(exp_id).version either. So, I just passed I over from the handler.

Reply:
OK. So, ExplorationStatsModel is cumulative until a particular version -- is that right?

If that's the case, that's fine -- please make sure that's documented in the model docstring (say something about this model instance containing stats from versions 1 to the version given in the key). And here, don't use "all" -- have the frontend pass in the version specifically, or have the frontend pass in nothing and, in the controller, get the version of the exploration.

Reply:
Done. We have a new handler now which doesn't take in the exp_version and uses the latest one for stats retrieval.